geometric/run_json.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+import json
+import geometric
+
+def get_qc_schema_traj(qc_schema_input, progress):
+    qc_schema_traj = []
+
+    return qc_schema_traj
+
+def parse_input_json_dict(in_json_dict):
+    """
+    Parse an input json dictionary into options, example:
+    in_json_dict = {
+        "schema_name": "qc_schema_optimization_input",
+        "schema_version", 1,
+        "geometric_options": {
+            "coordsys": "tric",
+            "conv": 1.e-7
+        }
+        "input_specification": qc_schema_input,
+    }
+    qc_schema_input = {
+        "schema_version": "v0.1",
+        "molecule": {
+            "geometry": [
+                0.0, 0.0, -0.1294769411935893, 0.0,
+               -1.494187339479985, 1.0274465079245698,
+                0.0, 1.494187339479985, 1.0274465079245698
+            ],
+            "symbols": ["O", "H", "H"],
+            "connectivity": [[0, 1, 1], [0, 2, 1]]
+        },
+        "driver": "gradient",
+        "model": {
+            "method": "UFF",
+            "basis": None
+        },
+        "keywords": {},
+        "program": "rdkit"
+    }
+    """
+    input_opts = in_json_dict['geometric_options'].copy()
+    input_specification = in_json_dict['input_specification'].copy()
+    # insert `fix_orientation` and `fix_com`
+    input_specification['molecule'].update({
+        'fix_orientation': True,
+        'fix_com': True
+    })
+    # Here we force the use of qcengine because other engines don't support qc schema
+    input_opts.update({
+        'qcengine': True,
+        'qcschema': input_specification
+    })
+    return input_opts
+
+def get_output_json_dict(in_json_dict, schema_traj):
+    # copy the input json data
+    out_json_dict = in_json_dict.copy()
+    out_json_dict.update({
+        "trajectory": schema_traj,
+        "energies": [s['properties']['return_energy'] for s in schema_traj],
+        "final_molecule": schema_traj[-1]
+    })
+    return out_json_dict
+
+def geometric_run_json(in_json_dict):
+    """ Take a input dictionary loaded from json, and return an output dictionary for json """
+    input_opts = parse_input_json_dict(in_json_dict)
```

Comment:
Best to do a `copy.deepcopy` when receiving this data. I think you might accidentally mutate the input with this function.

Reply:
The copy is done in `parse_input_json_dict()` function, but since the input dictionary shouldn't be too large, I will add another deepcopy here just to be safe.

Reply:
Just deep in mind `dict.copy()` is a shallow copy, not a deep copy. Since you are explicitly using `update` on future dictionaries there are possibilities of conflicts.